javascript/new-src/src/js/app.js

```diff
@@ -14,665 +14,698 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 angular
- .module('olaApp', ['ngRoute'])
- .constant('OLA', {
-  'MIN_CHANNEL_NUMBER': 1,
-  'MAX_CHANNEL_NUMBER': 512,
-  'MIN_CHANNEL_VALUE': 0,
-  'MAX_CHANNEL_VALUE': 255
- })
- .factory('$ola', ['$http', '$window', 'OLA',
-  function ($http, $window, OLA) {
-   'use strict';
-   // once olad supports json post data postEncode can go away
-   // and the header in post requests too.
-   var postEncode = function (data) {
-    var PostData = [];
-    for (var key in data) {
-     if (key === 'd' ||
-         key === 'remove_ports' ||
-         key === 'modify_ports' ||
-         key === 'add_ports') {
-      // this is here so dmx posts don't get broken because of removed comma's
-      PostData.push(key + '=' + data[key]);
-     } else {
-      PostData.push(key + '=' + encodeURIComponent(data[key]));
-     }
-    }
-    return PostData.join('&');
-   };
-   var dmxConvert = function (dmx) {
-    var length = 1,
-     integers = [];
-    for (var i = 0; i < OLA.MAX_CHANNEL_NUMBER; i++) {
-     integers[i] = parseInt(dmx[i], 10);
-     if (integers[i] > OLA.MIN_CHANNEL_VALUE) {
-      length = (i + 1);
-     }
+  .module('olaApp', ['ngRoute'])
+  .constant('OLA', {
+    'MIN_CHANNEL_NUMBER': 1,
+    'MAX_CHANNEL_NUMBER': 512,
+    'MIN_CHANNEL_VALUE': 0,
+    'MAX_CHANNEL_VALUE': 255
+  })
+  .factory('$ola', ['$http', '$window', 'OLA',
+    function($http, $window, OLA) {
+      'use strict';
+      // once olad supports json post data postEncode can go away
+      // and the header in post requests too.
+      var postEncode = function(data) {
+        var PostData = [];
+        for (var key in data) {
+          if (key === 'd' ||
+            key === 'remove_ports' ||
+            key === 'modify_ports' ||
+            key === 'add_ports') {
+            // this is here so dmx posts don't get broken
+            // because of removed comma's
+            PostData.push(key + '=' + data[key]);
+          } else {
+            PostData.push(key + '=' + encodeURIComponent(data[key]));
+          }
+        }
+        return PostData.join('&');
+      };
+      var dmxConvert = function(dmx) {
+        var length = 1,
+          integers = [];
+        for (var i = 0; i < OLA.MAX_CHANNEL_NUMBER; i++) {
+          integers[i] = parseInt(dmx[i], 10);
+          if (integers[i] > OLA.MIN_CHANNEL_VALUE) {
+            length = (i + 1);
+          }
+        }
+        integers.length = length;
+        return integers.join(',');
+      };
+      return {
+        get: {
+          Debug: function() {
```

Comment:
You can remove this again now can't you? Now it's being fetched via server info.